src/components/Searchbar.js

```diff
@@ -1,15 +1,161 @@
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { makeStyles } from '@material-ui/core/styles';
 import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
 import ArrowDropUpIcon from '@material-ui/icons/ArrowDropUp';
 import constants from '../styling/Constants';
 import { TextInput, DateInput, StyledButton } from '../components';
+import { useHistory } from 'react-router-dom';
+
+function getInputVal(wrapId) {
+    return document.getElementById(wrapId)?.getElementsByTagName('input')[0]?.value;
+}
+
+function getLocal(key) {
+    return window.localStorage.getItem(key);
+}
+
+function saveLocalSearch() {
+    window.localStorage.setItem('location', getInputVal('location-input-wrap'));
+
+    let date = getInputVal('date-input-wrap');
+    if (date) date = new Date(date);
+    window.localStorage.setItem('dateFrom', date);
+
+    let dateFrom = getInputVal('date-from-wrap');
+    if (dateFrom) dateFrom = new Date(dateFrom);
+    window.localStorage.setItem('dateFrom', dateFrom); // override date if needed
+
+    let dateTo = getInputVal('date-to-wrap');
+    if (dateTo) dateTo = new Date(dateTo);
+    window.localStorage.setItem('dateTo', dateTo);
+
+    window.localStorage.setItem('subjectName', getInputVal('subject-name-wrap'));
+    window.localStorage.setItem('submitterName', getInputVal('submitter-name-wrap'));
+    window.localStorage.setItem('grade', getInputVal('grade-wrap'));
+    window.localStorage.setItem('teacher', getInputVal('teacher-wrap'));
+}
+
+function setElemVal(wrapId, val) {
+    document.getElementById(wrapId).getElementsByTagName('input').value = val;
+}
+
+function updateFields() {
+    setElemVal('location-input-wrap', getLocal('location'));
+    setElemVal('date-input-wrap', getLocal('dateFrom'));
+    setElemVal('date-from-wrap', getLocal('dateFrom'));
+    setElemVal('date-to-wrap', getLocal('dateTo'));
+    setElemVal('subject-name-wrap', getLocal('subjectName'));
+    setElemVal('submitter-name-wrap', getLocal('submitterName'));
+    setElemVal('school-wrap', getLocal('school'));
+    setElemVal('grade-wrap', getLocal('grade'));
+    setElemVal('teacher-wrap', getLocal('teacher'));
+}
+
+// execute search by redirecting to gallery with query params in URL
+function submitSearch(history, e) {
+    e.preventDefault();
+    saveLocalSearch();
+
+    // construct query string for URL path
+    const localFields = [
+        'location', 'dateFrom', 'dateTo',
+        'subjectName', 'submitterName', 'grade', 'teacher'
+    ];
+    
+    let localVals = {};
+    localFields.forEach((field) => {
+        localVals[field] = getLocal(field);
+        if (localVals[field] instanceof Date) {
+            localVals[field] = new Intl.DateTimeFormat('en', { 
+                month: 'short',
+                day: 'numeric',
```

Comment:
why is month short when everything else numeric

Reply:
function was motived to util/format.js in a later commit and i added comments to explain it better. this part is meant to format the date in human-readable format, e.g. "Jan 1, 1930" instead of "0 1, 1930" as would be the case for numeric month

Reply:
Ok makes sense - I would just caution that there might be some concerns with this in the future (there's a reason why most technologies/large companies use integer format for everything). But I'll approve for now!